GetIntoTeachingApi/Controllers/OperationsController.cs

```diff
@@ -77,5 +82,38 @@ public async Task<IActionResult> HealthCheck()
 
             return Ok(response);
         }
+
+        [HttpPut]
+        [Authorize(Roles = "Admin,Crm")]
+        [Route("pause_crm_integration")]
+        [SwaggerOperation(
+            Summary = "Temporarily pauses the integration with the CRM.",
+            Description = "The CRM is taken offline for updates occasionally; this can result " +
+            "in errors when the API attempts to call out to the CRM. The CRM can call this endpoint " +
+            "to pause the API -> CRM integration (if not manually resumed it will auto-resume in 6 hours).",
+            OperationId = "PauseCrmIntegration",
+            Tags = new[] { "Operations" })]
+        [ProducesResponseType(typeof(HealthCheckResponse), StatusCodes.Status204NoContent)]
+        public IActionResult PauseCrmIntegration()
+        {
+            _appSettings.CrmIntegrationPausedUntil = DateTime.UtcNow.AddHours(6);

```

Comment:
Would it be worth using an env var for `6` so that it can be adjusted as necessary?

Reply:
I think if it was likely to be changed then that would make sense, but I see it as something that should be static; essentially a value that is a fair bit greater than the maximum expected offline time for the CRM during an update and less than 24 hours (the maximum time we can hold on to PII for before we auto-purge it) -- I'll add a comment to make that clearer. It's also a safety net that hopefully won't come into effect if the CRM team follow protocol and call `/operations/resume_crm_integration` when the update is finished.

Reply:
Sure. I was also thinking about testing environments where it could be useful to lower it, but if you reckon there's no need that's fine. 👍